packages/database/src/migrations/20210201051217-SyncLaosEocEntitiesWithLaosSchools-modifies-data.js

```diff
@@ -0,0 +1,189 @@
+    },
+  },
+  Phongsaly: {
+    capital: {
+      code: 'LA_PROV_CAPITAL_PHONGSALI',
+      subDistrictCode: 'LA_Phongsaly District',
+    },
+  },
+  Xainyabuli: {
+    capital: {
+      code: 'LA_PROV_CAPITAL_XAINYABULI',
+      lat: '19.25,', // These were wrong in the last import
+      lon: '101.75',
+      subDistrictCode: 'LA_Xayabury District',
+    },
+  },
+  Salavan: {
+    capital: {
+      code: 'LA_PROV_CAPITAL_SALAVAN',
+      subDistrictCode: 'LA_Saravane District',
+    },
+  },
+  Savannakhet: {
+    capital: {
+      code: 'LA_PROV_CAPITAL_KAYSONE_PHOMVIHANE',
+      subDistrictCode: 'LA_KaysonePhomvihane District',
+    },
+  },
+  Xekong: {
+    capital: {
+      code: 'LA_PROV_CAPITAL_LAMARN',
+      subDistrictCode: 'LA_Lamarm District',
+    },
+  },
+  'Vientiane Capital': {
+    capital: {
+      code: 'LA_PROV_CAPITAL_SISATTANAK',
+      subDistrictCode: 'LA_Sisattanak District',
+    },
+  },
+  'Vientiane Province': {
+    capital: {
+      code: 'LA_PROV_CAPITAL_PHÔNHÔNG',
+      subDistrictCode: 'LA_Phonhong District',
+    },
+  },
+  Xiangkhouang: {
+    capital: {
+      code: 'LA_PROV_CAPITAL_PEK',
+      subDistrictCode: 'LA_Pek District',
+    },
+  },
+  Xaisomboun: {
+    capital: {
+      code: 'LA_PROV_CAPITAL_ANOUVONG',
+      subDistrictCode: 'LA_Anouvong District',
+    },
+  },
+};
+
+const getEntityId = async (db, code) => {
+  const results = await db.runSql(`SELECT id FROM entity WHERE code = '${code}';`);
+
+  if (results.rows.length > 0) {
+    return results.rows[0].id;
+  }
+
+  throw new Error('Entity not found');
+};
+
+const getEntityHierarchyId = async (db, name) => {
+  const results = await db.runSql(`SELECT id FROM entity_hierarchy WHERE name = '${name}';`);
+
+  if (results.rows.length > 0) {
+    return results.rows[0].id;
+  }
+
+  throw new Error('Entity hierarchy not found');
+};
+
+exports.up = async function (db) {
+  // 1. Update wrong coords for Xainyabuli
+  const point = {
+    type: 'Point',
+    coordinates: [PROVINCES.Xainyabuli.capital.lon, PROVINCES.Xainyabuli.capital.lat],
+  };
+  await db.runSql(
+    `UPDATE entity SET point = ST_Force2D(ST_GeomFromGeoJSON('${JSON.stringify(
+      point,
+    )}')) WHERE code = '${PROVINCES.Xainyabuli.capital.code}';`,
+  );
+
+  // 2. Move cities under sub_district
+  const laosEocEntityHierarchyId = await getEntityHierarchyId(db, 'laos_eoc');
+
+  for (const provinceName of Object.keys(PROVINCES)) {
+    const province = PROVINCES[provinceName];
+
+    const subDistrictId = await getEntityId(db, province.capital.subDistrictCode);
+
+    const cityId = await getEntityId(db, province.capital.code);
+
+    await db.runSql(`UPDATE entity SET parent_id = '${subDistrictId}' WHERE id = '${cityId}';`);
+
+    await db.runSql(`
+      INSERT INTO entity_relation (id, parent_id, child_id, entity_hierarchy_id)
+      VALUES (generate_object_id(), '${subDistrictId}', '${cityId}', '${laosEocEntityHierarchyId}');
```

Comment:
I guess we also need to delete existing entity relations?

Reply:
Ah very good spotting! I was messing with data manually and missed that step. Updated